metricbeat/module/system/process/process.go

```diff
@@ -124,9 +125,39 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 	}
 
 	for _, proc := range procs {
+		rootFields := common.MapStr{
+			"process": common.MapStr{
+				"name": getAndRemove(proc, "name"),
+				"pid":  getAndRemove(proc, "pid"),
+				"ppid": getAndRemove(proc, "ppid"),
+			},
+			"user": common.MapStr{
+				"name": getAndRemove(proc, "username"),
+			},
+		}
+
+		if cwd := getAndRemove(proc, "cwd"); cwd != nil {
+			rootFields.Put("process.working_directory", cwd)
+		}
+
+		if exe := getAndRemove(proc, "exe"); exe != nil {
+			rootFields.Put("process.executable", exe)
+		}
+
+		if args := getAndRemove(proc, "args"); args != nil {
+			rootFields.Put("process.args", args)
+		}
+
 		e := mb.Event{
+			RootFields:      rootFields,
 			MetricSetFields: proc,
 		}
 		r.Event(e)
 	}
 }
+
+func getAndRemove(from common.MapStr, field string) interface{} {
+	v := from[field]
```

Comment:
I think we should check here if the field actually exists. Somethink like:

```
if v, ok := from[field]; ok { 
   delete ...
```

Reply:
Ok, but the result is going to be the same :slightly_smiling_face:

Reply:
Doesn't your code panic in case the field is not set?